backend/src/main/kotlin/com/projector/common/mongo/MongoCollectionsConfig.kt

```diff
@@ -0,0 +1,35 @@
+package com.projector.common.mongo
+
+import com.mongodb.client.model.Indexes
+import com.projector.timerecords.repository.TimeRecordDocument
+import com.projector.timerecords.repository.TimeRecordsRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.DependsOn
+import org.springframework.data.domain.Sort.Direction
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.index.Index
+import javax.annotation.PostConstruct
+
+
+@Configuration
+class MongoCollectionsConfig(
+        private val mongoConfig:MongoConfig
+) {
+
+    @PostConstruct
+    fun initIndexes() {
+        val db = mongoConfig.getDatabase()
```

Comment:
i should probably write a common index definition class that takes in collection name and indexes , and iterates and then creates the index